include/robot_localization/navsat_transform.hpp

```diff
@@ -139,13 +139,13 @@ class NavSatTransform : public rclcpp::Node
    * @brief Converts the odometry data back to GPS and broadcasts it
    * @param[out] filtered_gps The NavSatFix message to prepare
    */
-  bool prepareFilteredGps(sensor_msgs::msg::NavSatFix & filtered_gps);
+  bool prepareFilteredGps(std::unique_ptr<sensor_msgs::msg::NavSatFix> & filtered_gps);
```

Comment:
The signatures of these function should probably use raw pointers `sensor_msgs::msg::NavSatFix *` and then the calls for them would be `prepareFilteredGps(msg.get())`. This is just is smart pointer convention.